homeassistant/components/esphome/light.py

```diff
@@ -220,8 +231,14 @@ async def async_turn_on(self, **kwargs: Any) -> None:
         if (transition := kwargs.get(ATTR_TRANSITION)) is not None:
             data["transition_length"] = transition
 
-        if (color_temp := kwargs.get(ATTR_COLOR_TEMP)) is not None:
+        if (color_temp_k := kwargs.get(ATTR_COLOR_TEMP_KELVIN)) is not None:
+            # Do not use kelvin_to_mired here to prevent precision loss
+            data["color_temperature"] = 1000000.0 / color_temp_k
+        elif (color_temp := kwargs.get(ATTR_COLOR_TEMP)) is not None:
```

Comment:
Untested (traveling), but I think you can drop everything but ATTR_COLOR_TEMP_KELVIN since the underlying light platform will already do the conversion to kelvin anytime the legacy ATTR_COLOR_TEMP is passed

Reply:
See preprocess_turn_on_alternatives in light/__init__.py

Reply:
https://github.com/home-assistant/core/blob/31f845bfe03aa6e8284deeb6d8f81f2effe473a7/homeassistant/components/light/__init__.py#L321

Reply:
Looking through parts of the template light class, it appears as though using kelvin will override any mireds properties we set manually. So, in theory, removing the ATTR_COLOR_TEMP shouldn't do much.

I did think of and test one thing: if we call the service `light.turn_on` and use mireds temperature, it will convert to Kelvin and back, which does lead to small rounding errors (less than a mired from my limited testing). In practice this _shouldn't_ make too much of a different, and I have to imagine the number of users who would be significantly impacted is vanishingly small, but it's something we should keep in mind. I'll go ahead and make the change, but we should make sure the impact is as minor as I think it is.

Reply:
Let's take it out.  We can fix the rounding on the light platform in a followup

Reply:
I double checked the frontend as well. Its using kelvin all the way though so only anyone still using mireds would be affected by the double conversion which is an existing issue so its seems ok

https://github.com/home-assistant/frontend/blob/eb552530e2514a51c995a4e750e8ceff8684342b/src/dialogs/more-info/components/lights/light-color-temp-picker.ts#L112